File: 0683/sesión 13 - MasterPage+CRUD/sin CRUD/SoftInv/SoftInvDBManager/DBManagerMySQL.cs
using System;
using System.Collections.Generic;
using System.Data.Common;
using System.Data.SqlClient;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using MySql.Data.MySqlClient;

namespace SoftInv.Db
{
    public class DBManagerMySQL: DBManager
    {
        private DBManagerMySQL()
        {
            //construtor ocultado que no se puedan crear instancias desde fuera de la clase
        }

        public static void CreateConcreteInstance()
        {
            if (DBManager.dbManager == null)
                DBManager.dbManager = new DBManagerMySQL();
        }

        protected override void RealizarConexionABaseDeDatos(string cadenaDeConexion)
        {
            this.conexion = new MySqlConnection(cadenaDeConexion);
        }

        protected override string ObtenerCadenaDeConexion()
        {
            MySqlConnectionStringBuilder csBuilder = new MySqlConnectionStringBuilder()
            {
                Server = $"{this.nombreDeHost},{this.puerto}",
                Database = this.baseDeDatos,
                UserID = this.usuario,
                Password = DBManager.Descifrar(this.contraseña)
            };

            return csBuilder.ConnectionString;
        }

        public override DbCommand CrearComando()
        {
            return new MySqlCommand();
        }

        public override string RetornarSQLParaUltimoAutoGenerado()
        {
            string sql = "SELECT @@LAST_INSERT_ID AS ID";
            return sql;
        }
    }
}

File: 0682/sesión 22 - ex1_2025_2_P3/Pregunta 03 - Estudiantes/Pregunta3_ProyectoBase/TransitSoftDBManager/src/main/java/pe/pucp/transitsoft/db/DBManagerMySQL.java
package pe.pucp.transitsoft.db;


public class DBManagerMySQL extends DBManager{
    
    protected DBManagerMySQL(){
        
    }

    @Override
    protected String getURL() {
        String url = this.tipo_de_driver.concat("://");
        url = url.concat(this.nombre_de_host);
        url = url.concat(":");
        url = url.concat(this.puerto);
        url = url.concat("/");
        url = url.concat(this.base_de_datos);
        return url;
    }
    
    @Override
    public String retornarSQLParaUltimoAutoGenerado(){
        String sql = "SELECT @@LAST_INSERT_ID AS ID";
        return sql;
    }
    
}
